src/olympia/editors/models.py

```diff
@@ -716,6 +716,9 @@ def all_users_by_score(cls):
 
         for row in query:
             user_id, name, total = row
+            if not name:
+                name = cls.objects.values_list('user__username', flat=True).get(user__id=user_id)
```

Comment:
It would be preferable to change the `_leaderboard_query()` to return the `username` when the `display_name` is blank using [conditional expressions](https://docs.djangoproject.com/en/1.8/ref/models/conditional-expressions/). Alternatively, a simpler approach could be to modify `_leaderboard_query()` to return both `username` and `display_name`, or even return proper `UserProfile` objects - the problem is then you'd need to modify every piece of code that uses that method to make it compatible with the changes.

Reply:
I thought of returning both `username` and `display_name` in `_leaderboard_query()` but it was being used in `get_leaderboards()` and not sure where else. 
So, this was the best approach I could've think of.